### PR TITLE
Added documentation for maintaining, extending and deploying githubIssueLabler

### PR DESCRIPTION
**Pull request by Anipik**

### CODE REVIEW

File: src/Microsoft.DotNet.Github.IssueLabeler/README.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+It is web api which automatically labels an issue when it is opened in a Github repo. This tool is currently deployed as azure web service for the [corefx](https://github.com/dotnet/corefx.git) repo.
+
+## How it works
+The web api uses a pretrained [Ml.Net](https://github.com/dotnet/machinelearning) model which is consumed through a nuget package Microsoft.DotNet.GitHubIssueLabeler.Assets. This model has been trained on around 10,000 issues already labeled in the corefx repo. The code for the model is available [here](https://github.com/dotnet/machinelearning-samples/tree/master/samples/csharp/end-to-end-apps/MulticlassClassification-GitHubLabeler).
+
+Whenever an issue is opened in the repo, the web api receives the payload (using webhooks) containing all the information about the issue like title, body, milestone, issue number etc. It then supplies this information to already loaded pretrained model and the model predicts a probability distribution over the all possible labels. We then take the label with maximum probablity and compare it with a threshold. if the predicted probablity is greater than threshold then we apply the label otherwise we do nothing.
```
Comment (danmosemsft) [inline src/Microsoft.DotNet.Github.IssueLabeler/README.md]: tpyo -- probablity

Comment (danmosemsft) [reply src/Microsoft.DotNet.Github.IssueLabeler/README.md]: x 2

File: src/Microsoft.DotNet.Github.IssueLabeler/README.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+## How to deploy\maintain the web api
+In order to deploy this web api, you need to publish it as an azure service. After publishing the app, you should enable the logging. This can be done in 2 ways
+- You can add a web.config to your project. You require 
+```<aspNetCore processPath=".\Microsoft.DotNet.Github.IssueLabeler.exe" stdoutLogEnabled="true" stdoutLogFile="\\?\%home%\LogFiles\stdout" />``` in your web.config file.
+
+- You can enable logging by editing the web.config on your azure portal, created while publishing the web service.
+
+After the logging has been enabled, the api will generate logs at the path supplied in your web.config file. All the lines which contain the issue number and predicted label (with confidence) start with #. You can use this regex to clean the log file.
```
Comment (danmosemsft) [inline src/Microsoft.DotNet.Github.IssueLabeler/README.md]: "a regex" ?

File: src/Microsoft.DotNet.Github.IssueLabeler/README.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
+# Microsoft.DotNet.Gitthub.IssueLabeler
+
+It is web api which automatically labels an issue when it is opened in a Github repo. This tool is currently deployed as azure web service for the [corefx](https://github.com/dotnet/corefx.git) repo.
+
+## How it works
+The web api uses a pretrained [Ml.Net](https://github.com/dotnet/machinelearning) model which is consumed through a nuget package Microsoft.DotNet.GitHubIssueLabeler.Assets. This model has been trained on around 10,000 issues already labeled in the corefx repo. The code for the model is available [here](https://github.com/dotnet/machinelearning-samples/tree/master/samples/csharp/end-to-end-apps/MulticlassClassification-GitHubLabeler).
```
Comment (danmosemsft) [inline src/Microsoft.DotNet.Github.IssueLabeler/README.md]: Ml.NET -> ML.NET

File: src/Microsoft.DotNet.Github.IssueLabeler/README.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+In order to deploy this web api, you need to publish it as an azure service. After publishing the app, you should enable the logging. This can be done in 2 ways
+- You can add a web.config to your project. You require 
+```<aspNetCore processPath=".\Microsoft.DotNet.Github.IssueLabeler.exe" stdoutLogEnabled="true" stdoutLogFile="\\?\%home%\LogFiles\stdout" />``` in your web.config file.
+
+- You can enable logging by editing the web.config on your azure portal, created while publishing the web service.
+
+After the logging has been enabled, the api will generate logs at the path supplied in your web.config file. All the lines which contain the issue number and predicted label (with confidence) start with #. You can use this regex to clean the log file.
+
+## How to use this for other repos
+The only thing you need to use it for any other repo is a model which has been trained on the issues associated with that repo.
```
Comment (danmosemsft) [inline src/Microsoft.DotNet.Github.IssueLabeler/README.md]: Presuambly I have to edit the web.config to include the repo URL?

How do I set up the web hook?

Comment (Anipik) [reply src/Microsoft.DotNet.Github.IssueLabeler/README.md]: > Presuambly I have to edit the web.config to include the repo URL?

This is done in appsetting.json file. I added a couple of points to address this.

> How do I set up the web hook?

i added the server address that need to be added as webhook. Also added a link to the documentation on how to create a webhook in github

**Pull request merged**